### PR TITLE
now implemented the ability to list a space with a title

### PR DESCRIPTION
**Pull request by Managram**
connects #3

### CODE REVIEW

File: app/app.rb [hunk 1]
```diff
@@ -1,6 +1,26 @@
 require 'sinatra/base'
 
+ENV['RACK_ENV'] ||= 'development'
+
+require_relative 'models/data_mapper_setup'
+
+
```
Comment (junyuanxue) [inline app/app.rb:7]: extra blank line

File: app/app.rb [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
 class MakersBnb < Sinatra::Base
+
+  get '/space/new' do
+    erb(:"space/new")
+  end
+
+  post '/space/new' do
+    space = Space.new(title: params[:title])
```
Comment (rhiannonruth) [inline app/app.rb]: title -> name

File: app/models/space.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'data_mapper'
+
+class Space
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :title, String
```
Comment (rhiannonruth) [inline app/models/space.rb]: title -> name

File: app/views/space/new.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<form action="/space/new" method="post">
+  Space Title:<br>
```
Comment (junyuanxue) [inline app/views/space/new.erb]: space name

File: app/views/space/new.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+  <input type="text" name="title"><br>
```
Comment (rhiannonruth) [inline app/views/space/new.erb]: title -> name

File: spec/features/space_spec.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+feature 'creating a new space' do
+  scenario 'user creates a new space' do
+    visit '/space/new'
+    fill_in :title, with: "newspace"
```
Comment (rhiannonruth) [inline spec/features/space_spec.rb]: title -> name

File: spec/features/space_spec.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    click_button("create space")
+    expect(Space.first.title).to eq "newspace"
```
Comment (rhiannonruth) [inline spec/features/space_spec.rb]: title -> name

File: spec/spec_helper.rb [hunk 8]
```diff
@@ -5,12 +5,20 @@
...
 Capybara.app = MakersBnb
 
 RSpec.configure do |config|
-  
+
   config.include Capybara::DSL
+  config.before(:suite) do
+    DatabaseCleaner.strategy = :transaction
+    DatabaseCleaner.clean_with(:truncation)
+  end
```
Comment (rhiannonruth) [inline spec/spec_helper.rb]: missing: 

```
  config.before :each do
    DatabaseCleaner.start
  end

  config.after :each do
    DatabaseCleaner.clean
  end
```

**Comment by rhiannonruth** [comment]
:shipit:

**Pull request merged**